internal/storage/storager.go

```diff
@@ -0,0 +1,67 @@
+package storage
+
+import (
+	"context"
+	"os"
+
+	"github.com/Longreader/go-shortener-url.git/config"
+	"github.com/Longreader/go-shortener-url.git/internal/repository"
+	"github.com/Longreader/go-shortener-url.git/internal/repository/harddrive"
+	"github.com/Longreader/go-shortener-url.git/internal/repository/memory"
+	"github.com/Longreader/go-shortener-url.git/internal/repository/postgres"
+)
+
+// Storager - интерфейс для хранилища.
+type Storager interface {
+	Set( // Сократить ссылку.
+		ctx context.Context, url repository.URL, userID repository.User,
+	) (id repository.ID, err error)
+	Get( // Получить оригинальную ссылку по ID.
+		ctx context.Context, id repository.ID,
+	) (url repository.URL, err error)
+	GetAll( // Получить все ссылки пользователя.
+		ctx context.Context, user repository.User,
+	) (links []repository.LinkData, err error)
```

Comment:
```suggestion
	GetAllByUser( // Получить все ссылки пользователя.
		ctx context.Context, user repository.User,
	) (links []repository.LinkData, err error)
```
Так более точно и понятно. Иначе можно подумать, что метод возвращает вообще все записи.